services/src/main/java/org/fao/geonet/api/users/UsersApi.java

```diff
@@ -641,8 +649,12 @@ public void getUserIdenticon(
 
         PasswordEncoder encoder = PasswordUtil.encoder(ApplicationContextHolder.get());
 
-        if (!encoder.matches(passwordResetDto.getPasswordOld(), user.get().getPassword())) {
-            throw new IllegalArgumentException("The old password is not valid");
+        if (isUserAllowedToResetWithoutOldPassword(myProfile) == false
```

Comment:
With the change if the setting is `false`, the exception returned is `The old password is not valid.`, it seems a bit confusing.

Reply:
If the setting is false, the UI will not show the action for admin. Then it depends how much we want to highlight that settings. We could add a specific message for this case ?

Reply:
I was trying it with the API, maybe an specific message can help in this case, but probably not the most usual.